File: 5-11-2024_rotate90_gfg.py
def rotate(matrix):
    m = len(matrix)
    for i in range(m):
        for j in range(i, m):
            temp = matrix[i][j]
            matrix[i][j] = matrix[j][i]
            matrix[j][i] = temp
    for i in range(m):
        for j in range(m // 2):
            temp = matrix[i][j]
            matrix[i][j] = matrix[i][m - j - 1]
            matrix[i][m - j - 1] = temp

File: 5-11-2024_rotate90_gfg.cpp
void rotate(vector<vector<int> >& matrix) {
    int m=matrix.size();
    //int n=sizeof(matrix[0]);
    for(int i=0;i<m;i++){
        for(int j=i;j<m;j++){
            int temp=matrix[i][j];
            matrix[i][j]=matrix[j][i];
            matrix[j][i]=temp;
        }
    }
   for(int i=0;i<m;i++){
       for(int j=0;j<m/2;j++){
        int temp=matrix[i][j];
        matrix[i][j]=matrix[i][m-j-1];
        matrix[i][m-j-1]=temp;}
    }
     
   // return matrix;
}
